compiler/circle-mpqsolver/src/bisection/ErrorApproximator.test.cpp

```diff
@@ -0,0 +1,134 @@
+}
+
+class NConvGraph final : public SimpleGraph
+{
+protected:
+  void initInput(loco::Node *input) override
+  {
+    auto ci_input = loco::must_cast<luci::CircleNode *>(input);
+    ci_input->shape_status(luci::ShapeStatus::VALID);
+    auto qparam = std::make_unique<luci::CircleQuantParam>();
+    qparam->min.assign(_channel_size, _a_min);
+    qparam->max.assign(_channel_size, _a_max);
+    ci_input->quantparam(std::move(qparam));
+  }
+
+  loco::Node *insertGraphBody(loco::Node *input) override
+  {
+    _filter = _g->nodes()->create<luci::CircleConst>();
+    _filter->dtype(loco::DataType::FLOAT32);
+    _filter->shape({_channel_size, _f_w, _f_h, _channel_size});
+    _filter->shape_status(luci::ShapeStatus::VALID);
+    _filter->name("conv_filter");
+    uint32_t indices[4] = {
+      0,
+    };
+
+    uint32_t w_shape[4] = {_filter->dim(0).value(), _filter->dim(1).value(),
+                           _filter->dim(2).value(), _filter->dim(3).value()};
+
+    _filter->size<loco::DataType::FLOAT32>(w_shape[0] * w_shape[1] * w_shape[2] * w_shape[3]);
+
+    for (indices[0] = 0; indices[0] < w_shape[0]; ++indices[0])
+    {
+      for (indices[1] = 0; indices[1] < w_shape[1]; ++indices[1])
+      {
+        for (indices[2] = 0; indices[2] < w_shape[2]; ++indices[2])
+        {
+          for (indices[3] = 0; indices[3] < w_shape[3]; ++indices[3])
+          {
+            uint32_t offset = cal_offset(w_shape, indices);
+            _filter->at<loco::DataType::FLOAT32>(offset) = (offset % 2 == 0) ? _w_max : _w_min;
+          }
+        }
+      }
+    }
+
+    _bias = _g->nodes()->create<luci::CircleConst>();
+    _bias->dtype(loco::DataType::FLOAT32);
+    _bias->shape({_channel_size});
+    _bias->name("conv_bias");
+
+    _conv = _g->nodes()->create<luci::CircleConv2D>();
+    _conv->padding(luci::Padding::SAME);
+    _conv->fusedActivationFunction(luci::FusedActFunc::NONE);
+    _conv->dtype(loco::DataType::FLOAT32);
+    _conv->shape({1, _width, _height, _channel_size});
+    _conv->shape_status(luci::ShapeStatus::VALID);
+    _conv->name("conv");
+    _conv->filter(_filter);
+    _conv->bias(_bias);
+    _conv->input(input);
+
+    return _conv;
+  }
+
+public:
+  luci::CircleConv2D *_conv = nullptr;
+  luci::CircleConst *_filter = nullptr;
+  luci::CircleConst *_bias = nullptr;
+  uint32_t _f_w = 1;
+  uint32_t _f_h = 1;
+  float _w_min = -1.f;
+  float _w_max = 1.f;
+  float _a_min = -1.f;
+  float _a_max = 1.f;
+};
+
+} // namespace
+
+TEST(CircleMPQSolverErrorApproximatorTest, verifyResultsTest)
+{
+  NConvGraph g;
+  g.init();
+
+  auto value = mpqsolver::bisection::approximate(g._conv);
+  float expected = ((g._w_max - g._w_min) * g._channel_size * std::max(g._a_max, g._a_min) +
+                    (g._a_max - g._a_min) * g._channel_size * std::max(g._w_max, g._w_min)) *
+                   g._f_h * g._f_w * g._height * g._width * g._channel_size;
+  EXPECT_FLOAT_EQ(::fabs(value * 1.e+06f - expected), 0);
```

Comment:
```suggestion
  EXPECT_FLOAT_EQ(expected, value);
```

A recommended way is to place expected value first.